### PR TITLE
Enhance comments for SubscribableListener

### LINKED ISSUE #103833: Improve SubscribableListener's class level comment
[SubscribableListener's class level comment](https://github.com/elastic/elasticsearch/blob/9a7c62453f2be3d6627cfbeae505e8ed4544bc0f/server/src/main/java/org/elasticsearch/action/support/SubscribableListener.java#L31-L38) should include that once it is complete, `#addListener` will immediately set the result on any new subscribers. This is a very important feature of the class in order to use it properly in concurrent environments, but the important documentation is [hidden on the `#addListener` method](https://github.com/elastic/elasticsearch/blob/9a7c62453f2be3d6627cfbeae505e8ed4544bc0f/server/src/main/java/org/elasticsearch/action/support/SubscribableListener.java#L92-L95).

### PR DESCRIPTION
**Pull request by ankikuma**
This PR enhances class level comments for the SubscribableListener class.

Fixes issue #103833.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/support/SubscribableListener.java [hunk 1]
```diff
@@ -34,6 +34,10 @@
  * An {@link ActionListener} to which other {@link ActionListener} instances can subscribe, such that when this listener is completed it
  * fans-out its result to the subscribed listeners.
  * <p>
+ * If this listener is complete, {@link #addListener} completes the subscribing
```
Comment (DiannaHohensee) [inline server/src/main/java/org/elasticsearch/action/support/SubscribableListener.java:37]: The line lengths are a little uneven :) Could you rearrange things to wrap at the 130 character line limit?

Typically folks write comments out to 130 characters, or consistently shorter. This comment uses 130 already.

**Review by DiannaHohensee — APPROVED** [review]
LGTM, thanks for picking it up.

Looks like some setting is still a little off that's making the bot think you need to sign things 🤔 With Mikhail we had to fix getting an `external-contributor` label, but this looks like a different problem 🙂

File: server/src/main/java/org/elasticsearch/action/support/SubscribableListener.java (hunk 1 above)
Comment (DiannaHohensee) [reply server/src/main/java/org/elasticsearch/action/support/SubscribableListener.java]: Oops. Ankita was asking me about the other long lines, and I double checked the [contributor's guide](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md) and it's actually 140 characters 😅

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Pull request merged**